core/trino-main/src/main/java/io/trino/connector/ConnectorAwareNodeManager.java

```diff
@@ -42,7 +42,13 @@ public ConnectorAwareNodeManager(InternalNodeManager nodeManager, String environ
     @Override
     public Set<Node> getAllNodes()
     {
-        return ImmutableSet.copyOf(nodeManager.getActiveConnectorNodes(catalogName));
+        return ImmutableSet.<Node>builder()
+                .addAll(getWorkerNodes())
+                // append coordinators (before coordinators have connector
```

Comment:
Is that true for multi-coordinator setups? Can we judge about another coordinator here?

Reply:
This is how `getAllNodes` worked previously. AFAIK this should always be the case as other, inconsistent setups don't make much sense.

Reply:
I also changes it like this (instead of point fix for rubix) as community connectors might rely on this behavior (coordinators being part of get all nodes)

Reply:
limited to just current node (instead of all coordinators)